src/NetEscapades.AspNetCore.SecurityHeaders/Headers/ContentSecurityPolicy/CspDirectiveBuilderExtensions.cs

```diff
@@ -69,13 +72,35 @@ public static class CspDirectiveBuilderExtensions
         {
             if (string.IsNullOrWhiteSpace(uri))
             {
-                throw new System.ArgumentException("Uri may not be null or empty", nameof(uri));
+                throw new ArgumentException("Uri may not be null or empty", nameof(uri));
             }
 
             builder.Sources.Add(uri);
             return builder;
         }
 
+        /// <summary>
+        /// Allow resources from the given IEnumerable <paramref name="uris"/>. Elements may be any non-empty value
+        /// </summary>
+        /// <typeparam name="T">A <see cref="CspDirectiveBuilder"/> to configure.</typeparam>
+        /// <param name="builder">The <see cref="CspDirectiveBuilder"/> to apply the source to.</param>
+        /// <param name="uris">An IEnumerable of URIs to allow.</param>
+        /// <returns>The CSP builder for method chaining</returns>
+        public static T From<T>(this T builder, IEnumerable<string> uris) where T : CspDirectiveBuilder
+        {
+            if (uris == null || uris.Count() == 0)
```

Comment:
We shouldn't call `Count()` here, as it will enumerate `uris` twice. We should either remove the `uris.Count == 0` check (my preference) or we should switch to `IColleciton` instead